src/org/zaproxy/zap/spider/parser/SpiderHtmlFormParser.java

```diff
@@ -321,118 +318,19 @@ private void notifyPostResourceFound(HttpMessage message, int depth, String url,
-				String min = fc.getAttributesMap().get("min");
-				if (min != null) {
-					return min;
-				}
-				String max = fc.getAttributesMap().get("max");
-				if (max != null) {
-					return max;
-				}
-				return DEFAULT_NUMBER_VALUE;
-			}
-
-			if (type.equalsIgnoreCase("url")) {
-				return "http://www.example.com";
-			}
-
-			if (type.equalsIgnoreCase("email")) {
-				return "foo-bar@example.com";
-			}
-
-			if (type.equalsIgnoreCase("color")) {
-				return "#ffffff";
-			}
-
-			if (type.equalsIgnoreCase("tel")) {
-				return "9999999999";
-			}
-
-			if (type.equalsIgnoreCase("datetime")) {
-				SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
-				return format.format(getDefaultDate());
-			}
-			if (type.equalsIgnoreCase("datetime-local")) {
-				SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
-				return format.format(getDefaultDate());
-			}
-			if (type.equalsIgnoreCase("date")) {
-				SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
-				return format.format(getDefaultDate());
-			}
-			if (type.equalsIgnoreCase("time")) {
-				SimpleDateFormat format = new SimpleDateFormat("HH:mm:ss");
-				return format.format(getDefaultDate());
-			}
-			if (type.equalsIgnoreCase("month")) {
-				SimpleDateFormat format = new SimpleDateFormat("yyyy-MM");
-				return format.format(getDefaultDate());
-			}
-			if (type.equalsIgnoreCase("week")) {
-				SimpleDateFormat format = new SimpleDateFormat("yyyy-'W'ww");
-				return format.format(getDefaultDate());
-			}
-		} else if (fc.getFormControlType() == FormControlType.PASSWORD) {
-			return DEFAULT_PASS_VALUE;
-		} else if (fc.getFormControlType() == FormControlType.FILE) {
-			return DEFAULT_FILE_VALUE;
-		}
-		return DEFAULT_EMPTY_VALUE;
-	}
-
-	/**
-	 * Gets the default {@code Date}, to be used for default values of date fields.
-	 *
-	 * @return the date, never {@code null}.
-	 * @see #setDefaultDate(Date)
-	 */
-	private Date getDefaultDate() {
-		if (defaultDate == null) {
-			return new Date();
-		}
-		return defaultDate;
-	}
-
-	/**
-	 * Sets the default {@code Date}, to be used for default values of date fields.
-	 * <p>
-	 * If none ({@code null}) is set it's used the current date when generating the values for the fields.
-	 *
-	 * @param date the default date, might be {@code null}.
-	 * @since TODO add version
-	 */
-	public void setDefaultDate(Date date) {
-		this.defaultDate = date;
+		// Create an empty HashMap called 'fieldAttributes'
+		Map<String, String> fieldAttributes = new HashMap<String, String>();
+		
+		//Store all values in the FormFiled field into the Map 'fieldAttributes' 
+		fieldAttributes.putAll(field.getFormControl().getAttributesMap());
+		
+		// Places a key, Control Type, for each FormControlType
+		fieldAttributes.put("Control Type", field.getFormControl().getFormControlType().name());
+		return this.valueGenerator.getValue(null, null, fieldAttributes);		
```

Comment:
We have modified the SpiderHtmlFormParser to now return a URI to the DefaultValueGenerator instead of a null. We did this by creating a new variable of type URI and adding a line of code to the parsing method that stored the URI in our new variable. Would this approach be appropriate for supplying the URI to the ValueGenerator, or is there another way we should be returning it?